File: backend/src/main/java/com/example/backend/model/dto/BadgeDTO.java
package com.example.backend.model.dto;

public class BadgeDTO {
    private Long id;
    private String tooltip;
    private String icon;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTooltip() {
        return tooltip;
    }

    public void setTooltip(String tooltip) {
        this.tooltip = tooltip;
    }

    public String getIcon() {
        return icon;
    }

    public void setIcon(String icon) {
        this.icon = icon;
    }
}

File: frontend/src/app/shared/dto/badge.dto.ts
export class BadgeDto {
  public readonly id: number | null;
  public readonly tooltip: string;
  public readonly icon: string;

  constructor(
    id: number | null,
    tooltip: string,
    icon: string
  ) {
    this.id = id;
    this.tooltip = tooltip;
    this.icon = icon;
  }
}
